src/sql/src/plan/statement/ddl.rs

```diff
@@ -4897,6 +4898,53 @@ pub fn plan_alter_schema_rename(
     }))
 }
 
+pub fn plan_alter_schema_swap<F>(
+    scx: &mut StatementContext,
+    name_a: UnresolvedSchemaName,
+    name_b: UnresolvedSchemaName,
+    gen_temp_suffix: F,
+) -> Result<Plan, PlanError>
+where
+    F: Fn(&dyn Fn(&str) -> bool) -> Result<String, PlanError>,
+{
+    let schema_a = scx.resolve_schema(name_a.clone())?;
+    let schema_b = scx.resolve_schema(name_b.clone())?;
+
+    // Schemas need to be in the same database.
+    let db_spec = if schema_a.database() == schema_b.database() {
+        schema_a.database().clone()
+    } else {
+        sql_bail!("cannot swap schemas that are in different databases");
```

Comment:
Uncovered

Reply:
Wait, you should actually not be able to reference two different databases in a single connection,right? So this should be unreachable?

Reply:
Good call, this shouldn't be possible. I refactored the parser such that the object we're swapping is an `Ident` as opposed to a full `UnresolvedSchemaName`, which allowed me to refactor this case away

Reply:
We allow referencing objects from multiple databases in the same connection. We are like cockroach and unlike postgres in that way.

Reply:
Ooh, thanks for the pointer @mjibson
https://materialize.com/docs/sql/namespaces/#database-details